pkg/pillar/zedcloud/send.go

```diff
@@ -766,7 +773,38 @@ func SendOnIntf(workContext context.Context, ctx *ZedCloudContext, destURL strin
 				LocalAddr: &localTCPAddr,
 				Timeout:   clientConfig.TCPHandshakeTimeout,
 			}
-			transport.DialContext = d.DialContext
+			resolverCache := ctx.ResolverCacheFunc
+			dialWithResolverCache := func(ctx context.Context, network, addr string) (net.Conn, error) {
```

Comment:
Some comment for this function would be useful. From what I can tell it is really dialGivingPreferenceToResolverCache() in that it does a normal dial if the cache doesn't have the IP.

Reply:
Added comment - note that now it is moved into the global scope.